choose-index.md

```diff
@@ -29,19 +29,79 @@ Before introducing index selection, it is important to understand the ways TiDB
 
 ## Index selection rules
 
-TiDB provides a heuristic rule named skyline-pruning based on the cost estimation of each operator for accessing tables. It can reduce the probability of wrong index selection caused by wrong estimation.
+TiDB selects indexes based on rules or cost. the based rules include pre rules and Skyline-pruning. When selecting an index, TiDB tries the pre rule first. If the existing index satisfies a pre rule, TiDB will directly select the index. Otherwise, TiDB will use Skyline-Pruning to exclude unqualified indexes, and then based on the cost estimation of each operator for accessing tables, select the index with the lowest cost.
+
+### Selection based on rules
+
+#### Pre rules
+
+TiDB uses the following heuristic pre rules to select index:
+
++ rule 1: If the existing index satisfies "unique indexes with full match + no need to retrieve rows from a table (which means the plan generated by the index is IndexReader operator)", directly select this index.
+
++ rule 2: If the existing index satisfies "unique indexes + need to retrieve rows from a table (which means the plan generated by the index is IndexReader operator)", select the index with the smallest number of retrieve rows from a table as the candidate index.
+
++ rule 3: If the existing index satisfies "common indexes + no need to retrieve rows from a table + the number of rows to be read is less than the value of a certain threshold", select the index with the smallest number of rows to be read as the candidate index.
+
++ rule 4: If only one candidate index is selected based on rule 2 and 3, select this index. If two candidate index is separately selected based on rule 2 and 3, select the index with the smallest number of rows to be read (the number of rows with index + the number of rows to be retrieved from a table).
```

Comment:
If two candidate indexes are separately selected based on rule 2 and 3

Reply:
Since we compare two indexes, is it `the smaller` rather than `the smallest`?

Reply:
Thx! fixed in https://github.com/pingcap/docs/pull/6160/commits/ff5bc24bae4a5739cd0b3393e1c08f4ce92f65cb.